docs/architecture.md

```diff
@@ -40,6 +40,14 @@ Cumulus entity state data is stored in a set of [DynamoDB](https://aws.amazon.co
 
 Discovering data for ingest is handled via workflow step components using Cumulus `provider` and `collection` configurations and various triggers.    Data can be ingested from AWS S3, FTP, HTTPS and more.
 
+#### Database
+
+Cumulus utililzes a user-provided PostgreSQL database backend with data replication to an elasticsearch instance utilized for API search query efficency.   For legacy reasons, Core is currently also deploying a DynamoDB datastore that writes are replicated to in parallel with the PostgreSQL database writes.   These data stores will be removed in future releases.
```

Comment:
Open to feedback re: this, it's intended to be a true, merge-able, but not replacement for @laurenfrederick 's RDS arch doc work.

Reply:
```suggestion
Cumulus utilizes a user-provided PostgreSQL database backend. For improved API search query efficiency Cumulus provides data replication to an Elasticsearch instance.   For legacy reasons, Cumulus is currently also deploying a DynamoDB datastore, and writes are replicated in parallel with the PostgreSQL database writes.   The DynamoDB replicated tables and parallel writes will be removed in future releases.
```

Reply:
I had to read that first sentence a few times so reworded it. Then went on a rewording spree.